### PR TITLE
feat(ErrorSummary): ✨ Make the component

### LINKED ISSUE #1320: ErrorSummary: Develop React component

### PR DESCRIPTION
**Pull request by Barsnes**
Resolves #1320

### CODE REVIEW

**Comment by Barsnes** [comment]
@mimarz We should have a look at #1400 before we merge this, since this component uses the `List` component

File: packages/react/src/components/ErrorSummary/ErrorSummary.tsx
Comment (mimarz) [inline packages/react/src/components/ErrorSummary/ErrorSummary.tsx:19]: This is not needed though? New `List` is already handling this functionality for us.

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx
Comment (mimarz) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:14]: Suggest adding an code snippet example and explanation under "Bruk" chapter, like we have for `List`.

File: packages/react/src/components/ErrorSummary/ErrorSummaryHeading.tsx
Comment (mimarz) [inline packages/react/src/components/ErrorSummary/ErrorSummaryHeading.tsx:17]: Could this whole component not just be `export const ErrorSummaryHeading = List.Heading`? 😅  (also with new `displayName`)

**Review by mimarz — CHANGES_REQUESTED** [review]
Great work! I think we can do some more simplifications in terms of using `List`.

File: packages/react/src/components/ErrorSummary/ErrorSummary.tsx
Comment (Barsnes) [reply packages/react/src/components/ErrorSummary/ErrorSummary.tsx:19]: We need the id on the div that has `role="alert"`, and since the ID is sent only to `ErrorSummary.Heading`, we need this 😓

File: packages/react/src/components/ErrorSummary/ErrorSummaryHeading.tsx
Comment (Barsnes) [reply packages/react/src/components/ErrorSummary/ErrorSummaryHeading.tsx:17]: We need the id, so I don't think so?

File: packages/react/src/components/ErrorSummary/ErrorSummary.tsx
Comment (mimarz) [reply packages/react/src/components/ErrorSummary/ErrorSummary.tsx:19]: Doh! true true 🙈

**Review by mimarz — APPROVED** [review]
Fine work, good sir 🧐

**Pull request merged**
